components/alert/test/alert-toast.test.js

```diff
@@ -0,0 +1,147 @@
+import '../alert-toast.js';
+import { aTimeout, expect, fixture, html } from '@open-wc/testing';
+import { runConstructor } from '../../../tools/constructor-test-helper.js';
+
+const toastFixture = html`
+		<d2l-alert-toast id="button-close" type="default" button-text="Do it!" has-close-button
+			subtext="I am subtext. I'm here to test the wrapping capabilities of adding subtext to these alerts, as well as other styling issues. Feel free to add to me!">
+			A message.
+		</d2l-alert-toast>`;
+
+describe('d2l-alert-toast', () => {
+
+	let toast;
+	beforeEach(async() => {
+		toast = await fixture(toastFixture);
+	});
+
+	describe('accessibility', () => {
+
+		it('passes aXe tests', async() => {
+			toast.open = true;
+			await toast.updateComplete;
+			await expect(toast).to.be.accessible();
+		});
+
+		it('should have status role when open', async() => {
+			toast.open = true;
+			await toast.updateComplete;
+			expect(toast.getAttribute('role')).to.equal('status');
+		});
+
+	});
+
+	describe('constructor', () => {
+
+		it('should construct', () => {
+			runConstructor('d2l-alert-toast');
+		});
+
+	});
+
+	describe('state', () => {
+
+		it('should start closed', () => {
+			expect(toast._state).to.equal('closed');
+		});
+
+		it('should transition to preopening then opening when open is set to true', async() => {
+			expect(toast._state).to.equal('closed');
+			toast.open = true;
+			await toast.updateComplete;
+			expect(toast._state).to.equal('preopening');
+			await aTimeout(75);
+			expect(toast._state).to.equal('opening');
+		});
+
+		it('should transition from opening to open on animation end', async() => {
+
+			toast.open = true;
+			await toast.updateComplete;
+			toast._state = 'opening';
```

Comment:
I am a bit apprehensive about defining internals of the component states.  This _kind of_ tests the animation, but not really.  The internals are subject to change, but unfortunately we don't have a way (other than manual testing) to _really_ test the animation is working as expected.

Reply:
I had a lot of internal debate about whether to add them or not because it didn't feel great to be poking at internals but I ended up going for it because it felt worse to the state transitions at all. If we'd rather not poke at externals to this extent I can remove them.